WorkflowInstanceStreamProcessorTest.shouldNotTriggerBoundaryEventIfTaskIsCompleted is flaky

**Summary**

> Try to answer the following as best as possible

- How often does the test fail? so far seen only once
- Does it block your work? yes
- Do we suspect that it is a real failure? no

**Failures**

> Outline known failure cases, e.g. a failed assertion and its stacktrace obtained from Jenkins

<details><summary>Example assertion failure</summary>
<pre>
Error Message

Expecting:
  <[(COMMAND, CREATE),
    (EVENT, CREATED),
    (COMMAND, TRIGGER),
    (COMMAND, CANCEL),
    (COMMAND_REJECTION, TRIGGER),
    (COMMAND_REJECTION, CANCEL)]>
to contain exactly (and in same order):
  <[(COMMAND, CREATE),
    (EVENT, CREATED),
    (COMMAND, TRIGGER),
    (COMMAND_REJECTION, TRIGGER)]>
but some elements were not expected:
  <[(COMMAND, CANCEL), (COMMAND_REJECTION, CANCEL)]>

Stacktrace

java.lang.AssertionError: 

Expecting:
  <[(COMMAND, CREATE),
    (EVENT, CREATED),
    (COMMAND, TRIGGER),
    (COMMAND, CANCEL),
    (COMMAND_REJECTION, TRIGGER),
    (COMMAND_REJECTION, CANCEL)]>
to contain exactly (and in same order):
  <[(COMMAND, CREATE),
    (EVENT, CREATED),
    (COMMAND, TRIGGER),
    (COMMAND_REJECTION, TRIGGER)]>
but some elements were not expected:
  <[(COMMAND, CANCEL), (COMMAND_REJECTION, CANCEL)]>

	at io.zeebe.engine.processor.workflow.WorkflowInstanceStreamProcessorTest.shouldNotTriggerBoundaryEventIfTaskIsCompleted(WorkflowInstanceStreamProcessorTest.java:441)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:288)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:282)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)

</pre>
</details>

**Hypotheses**

> List any hypotheses if you have one; can be ommitted

**Logs**

> If possible, provide more context here, e.g. standard output logs, link to build, etc.

<details><summary>Logs</summary>
<pre>
16:47:19.083 [] INFO  io.zeebe.test - Test finished: shouldRejectCancellationInDirectSuccession(io.zeebe.engine.processor.workflow.WorkflowInstanceStreamProcessorTest)
16:47:19.083 [] INFO  io.zeebe.test - Test started: shouldNotTriggerBoundaryEventIfTaskIsCompleted(io.zeebe.engine.processor.workflow.WorkflowInstanceStreamProcessorTest)
16:47:19.139 [Broker-0-LogStream-0] DEBUG io.zeebe.logstreams - Configured log appender back pressure at partition 0 as AppenderVegasCfg{initialLimit=1024, maxConcurrency=32768, alphaLimit=0.7, betaLimit=0.95}. Window limiting is disabled
16:47:19.142 [] DEBUG io.zeebe.logstreams.snapshot - Available snapshots: []
16:47:19.623 [] DEBUG io.zeebe.logstreams.snapshot - Opened database from '/tmp/junit4290539589323729526/stream-0/state/runtime'.
16:47:19.624 [Broker-0-StreamProcessor-0] DEBUG io.zeebe.logstreams - Recovering state of partition 0 from snapshot
16:47:19.624 [Broker-0-StreamProcessor-0] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.zeebe.el.impl.feel.MessagePackValueMapper@5350705a)), function-provider: io.zeebe.el.impl.feel.FeelFunctionProvider@5d978440, configuration: Configuration(false)]
16:47:19.624 [Broker-0-StreamProcessor-0] INFO  io.zeebe.logstreams - Recovered state of partition 0 from snapshot at position -1
16:47:19.625 [Broker-0-StreamProcessor-0] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.zeebe.el.impl.feel.MessagePackValueMapper@59a81b2a)), function-provider: io.zeebe.el.impl.feel.FeelFunctionProvider@1caa8e7b, configuration: Configuration(false)]
16:47:19.851 [] INFO  io.zeebe.broker.test - Test failed, following records where exported:
16:47:19.853 [] INFO  io.zeebe.broker.test - Records on partition 0:
			Position: 4294967296 Key: -1
				RecordMetadata{recordType=COMMAND, intentValue=255, intent=CREATE, requestStreamId=-2147483648, requestId=-1, protocolVersion=1, valueType=WORKFLOW_INSTANCE_CREATION, rejectionType=NULL_VAL, rejectionReason=}
				Value:
{"bpmnProcessId":"process","workflowKey":-1,"workflowInstanceKey":-1,"version":-1,"variables":"gA==","fetchVariables":[]}
			Position: 4294967480 Key: 1
				RecordMetadata{recordType=EVENT, intentValue=255, intent=ELEMENT_ACTIVATING, requestStreamId=-2147483648, requestId=-1, protocolVersion=1, valueType=WORKFLOW_INSTANCE, rejectionType=NULL_VAL, rejectionReason=}
				Value:
{"bpmnProcessId":"process","version":1,"workflowKey":123,"workflowInstanceKey":1,"elementId":"process","flowScopeKey":-1,"bpmnElementType":"PROCESS","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1}
			Position: 4294967744 Key: 2
				RecordMetadata{recordType=EVENT, intentValue=255, intent=CREATED, requestStreamId=-2147483648, requestId=-1, protocolVersion=1, valueType=WORKFLOW_INSTANCE_CREATION, rejectionType=NULL_VAL, rejectionReason=}
				Value:
{"bpmnProcessId":"process","workflowKey":123,"workflowInstanceKey":1,"version":1,"variables":"gA==","fetchVariables":[]}
			Position: 4294967944 Key: 1
				RecordMetadata{recordType=EVENT, intentValue=255, intent=ELEMENT_ACTIVATED, requestStreamId=-2147483648, requestId=-1, protocolVersion=1, valueType=WORKFLOW_INSTANCE, rejectionType=NULL_VAL, rejectionReason=}
				Value:
{"bpmnProcessId":"process","version":1,"workflowKey":123,"workflowInstanceKey":1,"elementId":"process","flowScopeKey":-1,"bpmnElementType":"PROCESS","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1}
			Position: 4294968224 Key: 3
				RecordMetadata{recordType=EVENT, intentValue=255, intent=ELEMENT_ACTIVATING, requestStreamId=-2147483648, requestId=-1, protocolVersion=1, valueType=WORKFLOW_INSTANCE, rejectionType=NULL_VAL, rejectionReason=}
				Value:
{"bpmnProcessId":"process","version":1,"workflowKey":123,"workflowInstanceKey":1,"elementId":"startEvent_38db8c9b-f039-4662-9617-9e24fff5ba5d","flowScopeKey":1,"bpmnElementType":"START_EVENT","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1}
			Position: 4294968544 Key: 3
				RecordMetadata{recordType=EVENT, intentValue=255, intent=ELEMENT_ACTIVATED, requestStreamId=-2147483648, requestId=-1, protocolVersion=1, valueType=WORKFLOW_INSTANCE, rejectionType=NULL_VAL, rejectionReason=}
				Value:
{"bpmnProcessId":"process","version":1,"workflowKey":123,"workflowInstanceKey":1,"elementId":"startEvent_38db8c9b-f039-4662-9617-9e24fff5ba5d","flowScopeKey":1,"bpmnElementType":"START_EVENT","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1}
			Position: 4294968864 Key: 3
				RecordMetadata{recordType=EVENT, intentValue=255, intent=ELEMENT_COMPLETING, requestStreamId=-2147483648, requestId=-1, protocolVersion=1, valueType=WORKFLOW_INSTANCE, rejectionType=NULL_VAL, rejectionReason=}
				Value:
{"bpmnProcessId":"process","version":1,"workflowKey":123,"workflowInstanceKey":1,"elementId":"startEvent_38db8c9b-f039-4662-9617-9e24fff5ba5d","flowScopeKey":1,"bpmnElementType":"START_EVENT","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1}
			Position: 4294969184 Key: 3
				RecordMetadata{recordType=EVENT, intentValue=255, intent=ELEMENT_COMPLETED, requestStreamId=-2147483648, requestId=-1, protocolVersion=1, valueType=WORKFLOW_INSTANCE, rejectionType=NULL_VAL, rejectionReason=}
				Value:
{"bpmnProcessId":"process","version":1,"workflowKey":123,"workflowInstanceKey":1,"elementId":"startEvent_38db8c9b-f039-4662-9617-9e24fff5ba5d","flowScopeKey":1,"bpmnElementType":"START_EVENT","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1}
			Position: 4294969504 Key: 4
				RecordMetadata{recordType=EVENT, intentValue=255, intent=SEQUENCE_FLOW_TAKEN, requestStreamId=-2147483648, requestId=-1, protocolVersion=1, valueType=WORKFLOW_INSTANCE, rejectionType=NULL_VAL, rejectionReason=}
				Value:
{"bpmnProcessId":"process","version":1,"workflowKey":123,"workflowInstanceKey":1,"elementId":"sequenceFlow_ec403f54-ea77-4954-8956-5501ddac9fa1","flowScopeKey":1,"bpmnElementType":"SEQUENCE_FLOW","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1}
			Position: 4294969832 Key: 5
				RecordMetadata{recordType=EVENT, intentValue=255, intent=ELEMENT_ACTIVATING, requestStreamId=-2147483648, requestId=-1, protocolVersion=1, valueType=WORKFLOW_INSTANCE, rejectionType=NULL_VAL, rejectionReason=}
				Value:
{"bpmnProcessId":"process","version":1,"workflowKey":123,"workflowInstanceKey":1,"elementId":"task1","flowScopeKey":1,"bpmnElementType":"SERVICE_TASK","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1}
			Position: 4294970112 Key: -1
				RecordMetadata{recordType=COMMAND, intentValue=255, intent=CREATE, requestStreamId=-2147483648, requestId=-1, protocolVersion=1, valueType=TIMER, rejectionType=NULL_VAL, rejectionReason=}
				Value:
{"elementInstanceKey":5,"workflowInstanceKey":1,"dueDate":1587660440646,"targetElementId":"timer1","repetitions":1,"workflowKey":123}
			Position: 4294970304 Key: 5
				RecordMetadata{recordType=EVENT, intentValue=255, intent=ELEMENT_ACTIVATED, requestStreamId=-2147483648, requestId=-1, protocolVersion=1, valueType=WORKFLOW_INSTANCE, rejectionType=NULL_VAL, rejectionReason=}
				Value:
{"bpmnProcessId":"process","version":1,"workflowKey":123,"workflowInstanceKey":1,"elementId":"task1","flowScopeKey":1,"bpmnElementType":"SERVICE_TASK","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1}
			Position: 4294970592 Key: 6
				RecordMetadata{recordType=EVENT, intentValue=255, intent=CREATED, requestStreamId=-2147483648, requestId=-1, protocolVersion=1, valueType=TIMER, rejectionType=NULL_VAL, rejectionReason=}
				Value:
{"elementInstanceKey":5,"workflowInstanceKey":1,"dueDate":1587660440646,"targetElementId":"timer1","repetitions":1,"workflowKey":123}
			Position: 4294970800 Key: -1
				RecordMetadata{recordType=COMMAND, intentValue=255, intent=CREATE, requestStreamId=-2147483648, requestId=-1, protocolVersion=1, valueType=JOB, rejectionType=NULL_VAL, rejectionReason=}
				Value:
{"deadline":-1,"worker":"","retries":3,"type":"type","customHeaders":[packed value (length=1)],"variables":"gA==","errorMessage":"","errorCode":"","bpmnProcessId":"process","workflowDefinitionVersion":1,"workflowKey":123,"workflowInstanceKey":1,"elementId":"task1","elementInstanceKey":5}
			Position: 4294971112 Key: 7
				RecordMetadata{recordType=EVENT, intentValue=255, intent=CREATED, requestStreamId=-2147483648, requestId=-1, protocolVersion=1, valueType=JOB, rejectionType=NULL_VAL, rejectionReason=}
				Value:
{"deadline":-1,"worker":"","retries":3,"type":"type","customHeaders":[packed value (length=1)],"variables":"gA==","errorMessage":"","errorCode":"","bpmnProcessId":"process","workflowDefinitionVersion":1,"workflowKey":123,"workflowInstanceKey":1,"elementId":"task1","elementInstanceKey":5}
			Position: 4294971424 Key: 7
				RecordMetadata{recordType=EVENT, intentValue=255, intent=COMPLETED, requestStreamId=-2147483648, requestId=-1, protocolVersion=1, valueType=JOB, rejectionType=NULL_VAL, rejectionReason=}
				Value:
{"deadline":-1,"worker":"","retries":3,"type":"type","customHeaders":[packed value (length=1)],"variables":"gA==","errorMessage":"","errorCode":"","bpmnProcessId":"process","workflowDefinitionVersion":1,"workflowKey":123,"workflowInstanceKey":1,"elementId":"task1","elementInstanceKey":5}
			Position: 4294971720 Key: 5
				RecordMetadata{recordType=EVENT, intentValue=255, intent=ELEMENT_COMPLETING, requestStreamId=-2147483648, requestId=-1, protocolVersion=1, valueType=WORKFLOW_INSTANCE, rejectionType=NULL_VAL, rejectionReason=}
				Value:
{"bpmnProcessId":"process","version":1,"workflowKey":123,"workflowInstanceKey":1,"elementId":"task1","flowScopeKey":1,"bpmnElementType":"SERVICE_TASK","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1}
			Position: 4294972000 Key: 6
				RecordMetadata{recordType=COMMAND, intentValue=255, intent=TRIGGER, requestStreamId=-2147483648, requestId=-1, protocolVersion=1, valueType=TIMER, rejectionType=NULL_VAL, rejectionReason=}
				Value:
{"elementInstanceKey":5,"workflowInstanceKey":1,"dueDate":1587660440646,"targetElementId":"timer1","repetitions":1,"workflowKey":123}
			Position: 4294972192 Key: 6
				RecordMetadata{recordType=COMMAND, intentValue=255, intent=CANCEL, requestStreamId=-2147483648, requestId=-1, protocolVersion=1, valueType=TIMER, rejectionType=NULL_VAL, rejectionReason=}
				Value:
{"elementInstanceKey":5,"workflowInstanceKey":1,"dueDate":1587660440646,"targetElementId":"timer1","repetitions":1,"workflowKey":123}
			Position: 4294972384 Key: 5
				RecordMetadata{recordType=EVENT, intentValue=255, intent=ELEMENT_COMPLETED, requestStreamId=-2147483648, requestId=-1, protocolVersion=1, valueType=WORKFLOW_INSTANCE, rejectionType=NULL_VAL, rejectionReason=}
				Value:
{"bpmnProcessId":"process","version":1,"workflowKey":123,"workflowInstanceKey":1,"elementId":"task1","flowScopeKey":1,"bpmnElementType":"SERVICE_TASK","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1}
			Position: 4294972672 Key: 6
				RecordMetadata{recordType=COMMAND_REJECTION, intentValue=255, intent=TRIGGER, requestStreamId=-2147483648, requestId=-1, protocolVersion=1, valueType=TIMER, rejectionType=INVALID_STATE, rejectionReason=Expected to trigger a timer with key '6', but the timer is not active anymore}
				Value:
{"elementInstanceKey":5,"workflowInstanceKey":1,"dueDate":1587660440646,"targetElementId":"timer1","repetitions":1,"workflowKey":123}
			Position: 4294972960 Key: 6
				RecordMetadata{recordType=COMMAND_REJECTION, intentValue=255, intent=CANCEL, requestStreamId=-2147483648, requestId=-1, protocolVersion=1, valueType=TIMER, rejectionType=NOT_FOUND, rejectionReason=Expected to cancel timer with key '6', but no such timer was found}
				Value:
{"elementInstanceKey":5,"workflowInstanceKey":1,"dueDate":1587660440646,"targetElementId":"timer1","repetitions":1,"workflowKey":123}
			Position: 4294973232 Key: 9
				RecordMetadata{recordType=EVENT, intentValue=255, intent=SEQUENCE_FLOW_TAKEN, requestStreamId=-2147483648, requestId=-1, protocolVersion=1, valueType=WORKFLOW_INSTANCE, rejectionType=NULL_VAL, rejectionReason=}
				Value:
{"bpmnProcessId":"process","version":1,"workflowKey":123,"workflowInstanceKey":1,"elementId":"sequenceFlow_aa9ff9ec-6f4d-4607-90be-6aff91f29a14","flowScopeKey":1,"bpmnElementType":"SEQUENCE_FLOW","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1}
			Position: 4294973560 Key: 10
				RecordMetadata{recordType=EVENT, intentValue=255, intent=ELEMENT_ACTIVATING, requestStreamId=-2147483648, requestId=-1, protocolVersion=1, valueType=WORKFLOW_INSTANCE, rejectionType=NULL_VAL, rejectionReason=}
				Value:
{"bpmnProcessId":"process","version":1,"workflowKey":123,"workflowInstanceKey":1,"elementId":"end","flowScopeKey":1,"bpmnElementType":"END_EVENT","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1}
			Position: 4294973832 Key: 10
				RecordMetadata{recordType=EVENT, intentValue=255, intent=ELEMENT_ACTIVATED, requestStreamId=-2147483648, requestId=-1, protocolVersion=1, valueType=WORKFLOW_INSTANCE, rejectionType=NULL_VAL, rejectionReason=}
				Value:
{"bpmnProcessId":"process","version":1,"workflowKey":123,"workflowInstanceKey":1,"elementId":"end","flowScopeKey":1,"bpmnElementType":"END_EVENT","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1}
			Position: 4294974104 Key: 10
				RecordMetadata{recordType=EVENT, intentValue=255, intent=ELEMENT_COMPLETING, requestStreamId=-2147483648, requestId=-1, protocolVersion=1, valueType=WORKFLOW_INSTANCE, rejectionType=NULL_VAL, rejectionReason=}
				Value:
{"bpmnProcessId":"process","version":1,"workflowKey":123,"workflowInstanceKey":1,"elementId":"end","flowScopeKey":1,"bpmnElementType":"END_EVENT","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1}
			Position: 4294974376 Key: 10
				RecordMetadata{recordType=EVENT, intentValue=255, intent=ELEMENT_COMPLETED, requestStreamId=-2147483648, requestId=-1, protocolVersion=1, valueType=WORKFLOW_INSTANCE, rejectionType=NULL_VAL, rejectionReason=}
				Value:
{"bpmnProcessId":"process","version":1,"workflowKey":123,"workflowInstanceKey":1,"elementId":"end","flowScopeKey":1,"bpmnElementType":"END_EVENT","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1}
			Position: 4294974648 Key: 1
				RecordMetadata{recordType=EVENT, intentValue=255, intent=ELEMENT_COMPLETING, requestStreamId=-2147483648, requestId=-1, protocolVersion=1, valueType=WORKFLOW_INSTANCE, rejectionType=NULL_VAL, rejectionReason=}
				Value:
{"bpmnProcessId":"process","version":1,"workflowKey":123,"workflowInstanceKey":1,"elementId":"process","flowScopeKey":-1,"bpmnElementType":"PROCESS","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1}
			Position: 4294974928 Key: 1
				RecordMetadata{recordType=EVENT, intentValue=255, intent=ELEMENT_COMPLETED, requestStreamId=-2147483648, requestId=-1, protocolVersion=1, valueType=WORKFLOW_INSTANCE, rejectionType=NULL_VAL, rejectionReason=}
				Value:
{"bpmnProcessId":"process","version":1,"workflowKey":123,"workflowInstanceKey":1,"elementId":"process","flowScopeKey":-1,"bpmnElementType":"PROCESS","parentWorkflowInstanceKey":-1,"parentElementInstanceKey":-1}

16:47:19.854 [] DEBUG io.zeebe.util.buffer - Close stream processor
16:47:19.854 [Broker-0-StreamProcessor-0] DEBUG io.zeebe.logstreams - Closed stream processor controller Broker-0-StreamProcessor-0.
16:47:19.890 [Broker-0-LogStream-0] INFO  io.zeebe.logstreams - Close appender for log stream stream-0
16:47:19.891 [stream-0-write-buffer] DEBUG io.zeebe.dispatcher - Dispatcher closed
16:47:19.891 [Broker-0-LogStream-0] INFO  io.zeebe.logstreams - On closing logstream stream-0 close 14 readers
16:47:19.891 [Broker-0-LogStream-0] INFO  io.zeebe.logstreams - Close log storage with name stream-0
16:47:19.891 [] DEBUG io.zeebe.broker.test - Clean up test files on path /tmp/junit4290539589323729526
</pre>
</details>

--- a/engine/src/test/java/io/zeebe/engine/processor/workflow/WorkflowInstanceStreamProcessorTest.java
+++ b/engine/src/test/java/io/zeebe/engine/processor/workflow/WorkflowInstanceStreamProcessorTest.java
@@ -438,7 +438,7 @@ public final class WorkflowInstanceStreamProcessorTest {
     // then
     assertThat(envRule.events().onlyTimerRecords().collect(Collectors.toList()))
         .extracting(r -> tuple(r.getRecordType(), r.getIntent()))
-        .containsExactly(
+        .containsSubsequence(
             tuple(RecordType.COMMAND, TimerIntent.CREATE),
             tuple(RecordType.EVENT, TimerIntent.CREATED),
             tuple(RecordType.COMMAND, TimerIntent.TRIGGER),

